### PR TITLE
Initial integration

### PR DESCRIPTION
**Pull request by trust-no-one**

### CODE REVIEW

**Comment by aiwilliams** [comment]
So great to see another integration coming along! I hope we can get rid of more boilerplate code soon. Note that the Travis build is failing.

File: src/CbDefenseClient.ts [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        headers: {
+          "X-Auth-Token": `${config.apiKey}/${config.connectorId}`,
+        },
+      },
+      logger,
+    );
+  }
+
+  public async getAccountDetails(): Promise<CbDefenseAccount> {
+    this.logger.info("Fetching a single Cb Defense device sensor agent...");
```
[the remark this answers is not in the document]
Comment (trust-no-one) [reply src/CbDefenseClient.ts]: Changed to `logger.trace`

File: src/invocationValidator.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import {
+  IntegrationExecutionContext,
+  IntegrationInstanceConfigError,
+  IntegrationInvocationEvent,
+} from "@jupiterone/jupiter-managed-integration-sdk";
+import { CbDefenseIntegrationConfig } from "./types";
+
+export default async function invocationValidator(
```
[the remark this answers is not in the document]
Comment (trust-no-one) [reply src/invocationValidator.ts]: Added a check for `401` error as part of the client API call, and throw an `IntegrationInstanceAuthenticationError` if that's the case.

**Comment by trust-no-one** [comment]
@aiwilliams all comments should be resolved. pls check again.

**Pull request merged**